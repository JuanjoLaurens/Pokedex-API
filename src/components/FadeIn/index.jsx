import PropTypes from 'prop-types';
import React, { memo } from 'react';
import { motion, AnimatePresence } from 'framer-motion';

const FadeIn = ({ children }) => {
  return (
    <AnimatePresence>
      <motion.div
        initial={{ opacity: 0, y: -100 }}
        transition={{ duration: 1.2 }}
        animate={{ opacity: 1, y: 0 }}
      >
        {children}
      </motion.div>
    </AnimatePresence>
  );
};

FadeIn.propTypes = {
  children: PropTypes.node.isRequired,
};

export default memo(FadeIn);
