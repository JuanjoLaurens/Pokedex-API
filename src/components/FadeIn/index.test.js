import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import FadeIn from './index';

describe('FadeIn', () => {
  it('renders correctly according to snapshot', () => {
    const wrapper = shallow(<FadeIn>Content</FadeIn>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
