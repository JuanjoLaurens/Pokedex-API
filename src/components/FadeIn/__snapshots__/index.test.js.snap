// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeIn renders correctly according to snapshot 1`] = `
<AnimatePresence>
  <ForwardRef(MotionComponent)
    animate={
      Object {
        "opacity": 1,
        "y": 0,
      }
    }
    initial={
      Object {
        "opacity": 0,
        "y": -100,
      }
    }
    transition={
      Object {
        "duration": 0.5,
      }
    }
  >
    Content
  </ForwardRef(MotionComponent)>
</AnimatePresence>
`;
